### PR TITLE
Site migration: progress view add start time

### PR DESCRIPTION
**Pull request by andfinally**
Note: This change is behind a feature flag and is part of an in-progress larger project. For context, see pbkcP4-8-p2.

#### Changes proposed in this Pull Request

* Stores migration start time in localStorage, and shows it during a migration.
* The design for this view had the time string worded as "Migration started on Nov 15, 2019 at 12:38 pm". This is a US-style date format, but having a preposition between date and time doesn't lend itself to translation. It's also more efficient to store the timestamp as one string, so I've done it as shown.

image

#### Testing instructions

(Note: most of these instructions are general instructions for testing migration. The bolded steps are the ones specific to this change)

* **Check out D36604-code and follow the test instructions for that patch.**
* Check out this pull request and run Calypso locally.
* Viewing calypso.localhost:3000/migrate/, you should see a SiteSelector component allowing you to choose a site to migrate a Jetpack site into.
* When you select a site, you should see another SiteSelector component, listing all the Jetpack sites your account has access to.
* Selecting one of those sites should take you to a confirmation screen.
* Accept the confirmation. You should see a progress screen detailing the steps of the migration.
  * **The migration start time should be displayed above the progress bar.** 
  * **It should persist when you navigate away from the page and back, and when you refresh the page.**
* **After the migration, when you start a new migration, a new start time should be displayed.**

### CODE REVIEW

File: client/my-sites/migrate/section-migrate.jsx [hunk 1]
```diff
@@ -72,6 +73,32 @@ class SectionMigrate extends Component {
 		this.props.navigateToSelectedSourceSite( sourceSiteId );
 	};
 
+	getStartTime = () => {
+		if ( ! window.localStorage ) {
```
Comment (jonathansadowski) [inline client/my-sites/migrate/section-migrate.jsx]: If we're going to use `localStorage`, I wonder if we should use [lib/browser-storage](https://github.com/Automattic/wp-calypso/tree/81162c9e9c8c2be8e78cac04e4c8c8735963434a/client/lib/browser-storage) instead.

That being said, I have a feeling that we should probably return this value in the API response. See p1575924542382600-slack-importing

File: client/my-sites/migrate/section-migrate.jsx [hunk 2]
```diff
@@ -72,6 +73,32 @@ class SectionMigrate extends Component {
...
+	};
+
+	setStartTime = () => {
+		if ( ! window.localStorage ) {
+			return;
+		}
+		const localeSlug = getLocaleSlug();
+		const startTime = moment()
+			.locale( localeSlug )
+			.format( 'LT D MMMM' );
```
Comment (jonathansadowski) [inline client/my-sites/migrate/section-migrate.jsx]: This technique doesn't result in something that's really a localized string. For example, it results in "Migration started 2:50 PM 9 December" for me. This isn't a format that would ever really be seen in the US.

Maybe we should use `LLLL` or `llll` instead.

File: client/my-sites/migrate/section-migrate.jsx (hunk 1 above)
Comment (andfinally) [reply client/my-sites/migrate/section-migrate.jsx]: I think you're right about the API - I'll see about adding the value there.

File: client/my-sites/migrate/section-migrate.jsx (hunk 2 above)
Comment (andfinally) [reply client/my-sites/migrate/section-migrate.jsx]: Ah right! The `LLLL` and `llll` formats seem a bit excessively detailed – we could go with `lll`, which is `June 9 2014 9:32 PM` in my locale.

**Comment by andfinally** [comment]
Blocked till the API is returning the start time.

**Comment by mattsherman** [comment]
I'll review this.

**Comment by mattsherman** [comment]
Tested and works as expected. Code looks good.

**Comment by mattsherman** [comment]
Accepted.

**Pull request merged**